update/genesis/export.go

```diff
@@ -197,15 +198,40 @@ func (se *stateExport) exportTrie(key string, trie data.Trie) error {
 		return err
 	}
 
-	for address, buff := range leaves {
+	if accType == state.DataTrie {
+		return se.exportDataTries(leaves, accType, shId, fileName)
+	}
+
+	return se.exportAccountLeafs(leaves, accType, shId, fileName)
+}
+
+func (se *stateExport) exportDataTries(leafs map[string][]byte, accType state.Type, shId uint32, fileName string) error {
+	for address, buff := range leafs {
+		keyToExport := CreateAccountKey(accType, shId, address)
+		err := se.writer.Write(fileName, keyToExport, buff)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (se *stateExport) exportAccountLeafs(leafs map[string][]byte, accType state.Type, shId uint32, fileName string) error {
+	for address, buff := range leafs {
+		keyToExport := CreateAccountKey(accType, shId, address)
 		account, err := NewEmptyAccount(accType)
 		if err != nil {
 			log.Warn("error creating new account account", "address", address, "error", err)
 			continue
 		}
 		err = se.marshalizer.Unmarshal(account, buff)
 		if err != nil {
-			log.Warn("error unmarshaling account", "address", address, "error", err)
+			err = se.writer.Write(fileName, keyToExport, buff)
+			if err != nil {
+				return err
+			}
+			log.Trace("error unmarshaling account this is maybe a code", "address", address, "error", err)
```

Comment:
why would it fail on an account with code?

Reply:
Code is saved as a []byte in the leaf. When accessing the leaf there is no information whether it is a code / account. So first there is a try to unmarshal into an account, otherwise it must be code.